pkg/code-generator/codegen/templates/snapshot_emitter_template.go

```diff
@@ -260,13 +284,32 @@ func (c *{{ lower_camel .GoName }}Emitter) Snapshots(watchNamespaces []string, o
 {{- if .ClusterScoped }}
 			case {{ lower_camel .Name }}List := <- {{ lower_camel .Name }}Chan:
 				record()
+
+				stats.RecordWithTags(
```

Comment:
suggestion: we can make record take a type and namespace and move that logic there.
then you can reuse the snapshotIn metric (making ResourcesIn redundant)

if your intention is to deprecate snapshotIn then we can do both in the mean time. just add a comment about the deprecation.